ql/src/java/org/apache/hadoop/hive/ql/QueryStats.java

```diff
@@ -0,0 +1,34 @@
+package org.apache.hadoop.hive.ql;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * QueryStats.
+ *
+ */
+public class QueryStats {
+  public long queryStart;
+  public long queryEnd;
+  public String queryID;
+  public String queryString;
+  public String sessionID;
+  public String IPAddress;
+  public String username;
+  public String database;
+  public String mapReduceStatsDesc;
+  public String currentTimeStamp;
+  public Map<String, String> taskProgress;
+  public Map<String, QueryPlan> plansInfo;
+  public Map<String, MapRedStats> mapReduceStats;
+
+  public QueryStats (String queryID, String queryString, long start, long end) {
```

Comment:
do not pass `end` to the constructor since you won't know when this query ends when it starts.

Reply:
done